src/json.cr

```diff
@@ -61,6 +61,8 @@
 # typeof(value[0])      # => JSON::Any
 # value[0].as_i         # => 1
 # typeof(value[0].as_i) # => Int32
+# value[0].as_i?        # => 1
+# typeof(value[0].as_i?)# => Int32 | Nil
```

Comment:
I would show a case where this returns nil, otherwise the difference is not very clear to be honest